### PR TITLE
Run samples in java11

### PR DESCRIPTION
**Pull request by ignasi35**
This is #73 but with the branch on `upstream` so [`deploy` stages also kick in](https://github.com/lagom/lagom-samples/pull/73#issuecomment-507626879).

### CODE REVIEW

**Review by dwijnand — APPROVED** [review]
I have reservations on defining the build like that (https://github.com/lagom/lagom-samples/pull/73#discussion_r299416282), but I'll take it upon myself to work out and propose my changes.

**Comment by ignasi35** [comment]
The job `deploy Shopping cart example (java/sbt)` fails when run on Java 11.

<details>
<summary>error</summary>

```
Oops, cannot start the server.
@7ccg2a2ai: Cannot load module[Module [com.example.shoppingcart.impl.ShoppingCartModule] cannot be instantiated.]
	at play.api.inject.Modules$.constructModule(Module.scala:181)
	at play.api.inject.Modules$.$anonfun$locate$4(Module.scala:140)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:237)
	at scala.collection.immutable.HashSet$HashSet1.foreach(HashSet.scala:321)
	at scala.collection.immutable.HashSet$HashTrieSet.foreach(HashSet.scala:977)
	at scala.collection.immutable.HashSet$HashTrieSet.foreach(HashSet.scala:977)
	at scala.collection.immutable.HashSet$HashTrieSet.foreach(HashSet.scala:977)
	at scala.collection.TraversableLike.map(TraversableLike.scala:237)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:230)
	at scala.collection.AbstractSet.scala$collection$SetLike$$super$map(Set.scala:51)
	at scala.collection.SetLike.map(SetLike.scala:104)
	at scala.collection.SetLike.map$(SetLike.scala:104)
	at scala.collection.AbstractSet.map(Set.scala:51)
	at play.api.inject.Modules$.locate(Module.scala:135)
	at play.api.inject.guice.GuiceableModule$.loadModules(GuiceInjectorBuilder.scala:303)
	at play.api.inject.guice.GuiceApplicationBuilder$.$anonfun$$lessinit$greater$default$9$1(GuiceApplicationBuilder.scala:36)
	at play.api.inject.guice.GuiceApplicationBuilder.applicationModule(GuiceApplicationBuilder.scala:115)
	at play.api.inject.guice.GuiceBuilder.injector(GuiceInjectorBuilder.scala:201)
	at play.api.inject.guice.GuiceApplicationBuilder.build(GuiceApplicationBuilder.scala:156)
	at play.api.inject.guice.GuiceApplicationLoader.load(GuiceApplicationLoader.scala:22)
	at play.core.server.ProdServerStart$.start(ProdServerStart.scala:59)
	at play.core.server.ProdServerStart$.main(ProdServerStart.scala:31)
	at play.core.server.ProdServerStart.main(ProdServerStart.scala)
Caused by: java.lang.UnsupportedClassVersionError: com/example/shoppingcart/impl/ShoppingCartModule has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at play.api.inject.Modules$.$anonfun$locate$5(Module.scala:140)
	at play.api.inject.Modules$.constructModule(Module.scala:152)
	... 22 more
```

</details>

The problem is the compilation happens on a JDK11 but the execution on a JDK8. The funny thing is that the same issue doesn't manifest on the `(java/maven)` and the `(scala/sbt)` variants of the `shopping-cart` sample.

**Review by marcospereira — APPROVED** [review]
@ignasi35 we have a green build and Maven is now executing for both Java 8 & 11. We can do additional changes later. For now, I think this is good to merge.

File: shopping-cart/shopping-cart-java/pom.xml [hunk 1]
```diff
@@ -16,6 +16,28 @@
...
+        We are doing so because we want to test this sample using Java 8
+        and Java 11, which requires some additional configuration.
+     -->
+    <profiles>
+        <profile>
+            <id>java11</id>
+            <activation>
+                <!-- This will activate if the build runs using a jdk version number that matches the prefix given. -->
+                <jdk>11</jdk>
+            </activation>
```
Comment (ignasi35) [inline shopping-cart/shopping-cart-java/pom.xml:30]: TIL! Nice.

**Pull request merged**